libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/WayNameView.java

```diff
@@ -32,8 +34,23 @@ public WayNameView(@NonNull Context context, @Nullable AttributeSet attrs, int d
     initialize();
   }
 
+  @Override
+  public void setVisibility(int visibility) {
+    if (visibility == VISIBLE) {
+      visible = true;
+      if (TextUtils.isEmpty(wayNameText.getText())) {
+        visibility = INVISIBLE;
+      }
+    } else {
+      visible = false;
+    }
+
+    super.setVisibility(visibility);
+  }
+
   public void updateWayNameText(String wayNameText) {
     this.wayNameText.setText(wayNameText);
+    updateVisibility(visible);
```

Comment:
each time update way name text will update visibility by using this internal visible flag. basically we remember the visibility state and wait until there is a name to show.

this could fix the original #2032 issue, I have verified it.